### PR TITLE
JBFundingCycleMetadataResolver fixes + packing util method

### PR DESCRIPTION
**Pull request by dhyon**
Contains some minor fixes for JBFundingCycleMetadataResolver lib that were discovered while mocking the metadata objects for unit testing JBETHPaymentTerminalStore (I've omitted the contract tests from this PR for diff clarity). Also contains the packing test util method because other people might want to use it in their tests. 

On possible paths forward:

1. get this current PR merged, knowing it can handle general setting/retrieving of values (but unaware of gnarly edge cases), and unblock the test effort for contracts interacting with this library
2. continue working on JBETHPaymentTerminalStore
2b. -or- immediately follow up with a comprehensive suite of test cases for the JBFundingCycleMetadataResolver lib

### CODE REVIEW

**Review by mejango — APPROVED** [review]
yussss

File: test/helpers/utils.js [hunk 1]
```diff
@@ -32,13 +32,81 @@ export function daysFromNow(days) {
...
+  holdFees = 0, // boolean
+  useLocalBalanceForRedemptions = 0, // boolean
+  useDataSourceForPay = 0, // boolean
+  useDataSourceForRedeem = 0, // boolean
+  dataSource = 0, // address
+} = {}) {
+  const one = ethers.BigNumber.from(1);
+
+  // version 1 in the bits 0-7 (8 bits).
+  var packed = ethers.BigNumber.from(version);
```
Comment (odd-amphora) [inline test/helpers/utils.js:75]: nit) let

File: test/helpers/utils.js [hunk 2]
```diff
@@ -32,13 +32,81 @@ export function daysFromNow(days) {
...
+  // useLocalBalanceForRedemptions in bit 65.
+  if (useLocalBalanceForRedemptions) packed = packed.or(one.shl(65));
+  // use pay data source in bit 66.
+  if (useDataSourceForPay) packed = packed.or(one.shl(66));
+  // use redeem data source in bit 67.
+  if (useDataSourceForRedeem) packed = packed.or(one.shl(67));
+  // data source address in bits 68-227.
+  packed = packed.or(ethers.BigNumber.from(dataSource).shl(68));
+
+  return packed;
```
Comment (odd-amphora) [inline test/helpers/utils.js:111]: nit) return packed.or(eth...

File: test/helpers/utils.js [hunk 3]
```diff
@@ -32,13 +32,81 @@ export function daysFromNow(days) {
...
+  if (pauseRedeem) packed = packed.or(one.shl(58));
+  // pause mint in bit 59.
+  if (pauseMint) packed = packed.or(one.shl(59));
+  // pause mint in bit 60.
+  if (pauseBurn) packed = packed.or(one.shl(60));
+  // pause change token in bit 61.
+  if (allowChangeToken) packed = packed.or(one.shl(61));
+  // allow terminal migration in bit 62.
+  if (allowTerminalMigration) packed = packed.or(one.shl(62));
+  // allow controller migration in bit 63.
```
Comment (odd-amphora) [inline test/helpers/utils.js:98]: I feel all of the comments here clutter this fn and it's already self-documenting as is. up to you whether you want to keep them or not.

File: test/helpers/utils.js [hunk 4]
```diff
@@ -32,13 +32,81 @@ export function daysFromNow(days) {
...
\ No newline at end of file
+}
+
+/**
+ * Returns a mock FundingCyleMetadata packed into a BigNumber
+ * @summary Should mirror the bit logic in JBFundingCycleMetadataResolver.sol.
+ * @param {custom obj} e.g. packFundingCycleMetadata({ reservedRate: 3500, pausePay: 1})
+ * @return {ethers.BigNumber}
+ * @note Passing in an empty obj will use default values below
+ */
```
Comment (odd-amphora) [inline test/helpers/utils.js:52]: While you're in the neighborhood would you mind adding a similar doc to other fns? :)

**Review by odd-amphora — APPROVED** [review]

File: test/helpers/utils.js (hunk 4 above)
Comment (dhyon) [reply test/helpers/utils.js:52]: Community service
